### PR TITLE
fix(explore): timepicker show human readable string in pill

### LINKED ISSUE #12285: [Explore] Timepicker converts date range type on pill
#### How to reproduce the bug

1. Go to Explore view, from date picker, choose `Range Type`: Last, then choose `Last week`.
2. see `Time Range` display as `2020-12-29 <= col < 2021-01-05`.
3. The time range is correct, but it is **not** the option that user chosen: Last week.
4. The reason that we offered `Last` or `Previous` is these human readable time range is easier for user to get what the time range is. If user pick these Range Type, we should display as it. If user pick from custom type and pick same date, we should display the date.

#### Screenshots

Screen_Shot_2021-01-05_at_10_24_20_AM-2

### Environment

latest master

cc @junlincc @zhaoyongjie @zuzana-vej

### PR DESCRIPTION
**Pull request by zhaoyongjie**
### SUMMARY
fix #12285 
Backward compatible. the timepicker pill should show human-readable text.
- show human-readable text in common / previous calendar frame.
- move frontend datetime expression to backend

#### Before
image

#### After
1. common frame
image
2. previous frame
image

### TEST PLAN
Added UT

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/utils/core.py [hunk 1]
```diff
@@ -1497,6 +1497,25 @@ def get_since_until(
     if time_range and time_range.startswith("Next") and separator not in time_range:
         time_range = _relative_start + separator + time_range
 
+    if (
+        time_range
+        and time_range.startswith("previous calendar week")
+        and separator not in time_range
+    ):
+        time_range = "DATETRUNC(DATEADD(DATETIME('today'), -1, WEEK), WEEK) : LASTDAY(DATEADD(DATETIME('today'), -1, WEEK), WEEK)"  # pylint: disable=line-too-long
```
Comment (mistercrunch) [inline superset/utils/core.py:1505]: Wait, these SQL expressions won't work across database engines. I think that we should use python logic to come up with time constants here.

**Review by mistercrunch — CHANGES_REQUESTED** [review]

File: superset/utils/core.py (hunk 1 above)
Comment (mistercrunch) [reply superset/utils/core.py:1505]: I think we need to backtrack some of the work from b592cc7e73 as well here

Comment (mistercrunch) [reply superset/utils/core.py:1505]: Oh nevermind! it looked like SQL to me at first and dug deeper to find the `pyparsing` lib.

**Review by mistercrunch — APPROVED** [review]

File: superset/utils/core.py (hunk 1 above)
Comment (ktmud) [reply superset/utils/core.py:1505]: If I'm reading the code right, these do not seem to be SQL expressions. They are passed to [`datetime_eval`](https://github.com/airbnb/incubator-superset/blob/865beae3d8644bb5e9df156e5fea80c3ec0202a2/superset/utils/core.py#L1540) which will translate them into actual timestamps before sending to SQL queries.

File: superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx [hunk 2]
```diff
@@ -183,15 +183,25 @@ export default function DateFilterControl(props: DateFilterLabelProps) {
...
-        setValidTimeRange(true);
-      }
-    });
+    const valueToLower = value.toLowerCase();
+    if (
+      valueToLower.startsWith('last') ||
+      valueToLower.startsWith('next') ||
+      valueToLower.startsWith('previous')
+    ) {
+      setActualTimeRange(value);
```
Comment (ktmud) [inline superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx:192]: Should we translate the value displayed for label?

Comment (zhaoyongjie) [reply superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx:192]: good point, I will add t().

Comment (mistercrunch) [reply superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx:192]: I think if you add `t()` here the backend will receive the translated value, we probably need a map with `key:label` here

Comment (zhaoyongjie) [reply superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx:192]: value is a dynamic value, not very easy to create a Map handle this.

File: superset/utils/core.py (hunk 1 above)
Comment (zhaoyongjie) [reply superset/utils/core.py:1505]: I implemented the parser with `pyparsing`, which allows time_range value to be more flexible

File: superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx (hunk 2 above)
Comment (ktmud) [reply superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx:192]: But there are limited number of combinations, right?

**Pull request merged**